iib/workers/tasks/build.py

```diff
@@ -495,6 +495,114 @@ def opm_index_add(bundles, binary_image, request_id, from_index=None):
     )
 
 
+@app.task
+def handle_rm_request(operators, binary_image, request_id, from_index, add_arches=None):
+    """
+    Coordinate the the work needed to build the index image with the input bundles.
+
+    :param list operators: a list of strings representing the name of the operators to
+        remove from the index image.
+    :param str binary_image: the pull specification of the image where the opm binary gets copied
+        from.
+    :param int request_id: the ID of the IIB build request
+    :param str from_index: the pull specification of the image containing the index that the index
+        image build will be based from.
+    :param list add_arches: the list of arches to build in addition to the arches ``from_index`` is
+        currently built for; if ``from_index`` is ``None``, then this is used as the list of arches
+        to build the index image for
+    :raises iib.exceptions.IIBError: if the index image build fails.
+    """
+    prebuild_info = _prepare_request_for_build(binary_image, request_id, from_index, add_arches)
+
+    error_callback = failed_request_callback.s(request_id=request_id)
+    arches = prebuild_info['arches']
+    for arch in sorted(arches):
+        # The bundles are not resolved since these are stable tags, and references
+        # to a bundle image using a digest fails when using the opm command.
+        # from_index is not resolved because docker does not support digest references
+        opm_index_rm.apply_async(
+            args=[operators, prebuild_info['binary_image_resolved'], request_id, from_index],
+            link_error=error_callback,
+            queue=f'iib_{arch}',
+            routing_key=f'iib_{arch}',
+        )
+
+    if not _poll_request(request_id, arches):
+        log.error('Not finishing the request since one of the underlying builds failed')
+        return
+
+    post_build_from_index_resolved = _get_resolved_image(from_index)
+    if post_build_from_index_resolved != prebuild_info['from_index_resolved']:
```

Comment:
In a separate PR or commit, could you please add this to the `add` workflow? You could probably put this in a private function that can be used in both places.